test/legacy/selectTest.ts

```diff
@@ -42,131 +44,15 @@ describe('select', () => {
-            expect(state.key).toBe(fooStore.k);
-            state.key = 'newValue';
-        });
-
-        let newAction = select({
-            key: () => fooStore.k,
-        })(updateAction);
-
-        newAction();
-
-        expect(fooStore.k).toBe('newValue');
-    });
-
-    it('can update observable map', () => {
-        let fooStore = createStore('foo', {
-            map: map({ k: 'v' }),
-        })();
-
-        interface ActionState {
-            map: ObservableMap<string>;
-        }
-
-        let updateAction = action('update')((state?: ActionState) => {
-            state.map.set('k', 'newValue');
-        });
-
-        let newAction = select({
-            map: () => fooStore.map,
-        })(updateAction);
-
-        newAction();
 
-        expect(fooStore.map.get('k')).toBe('newValue');
-    });
-
-    it('can update observable arrays', () => {
-        let fooStore = createStore('foo', {
-            array: ['v'],
-        })();
-
-        interface ActionState {
-            array: string[];
-        }
-
-        let updateAction = action('update')((state?: ActionState) => {
-            state.array.push('newValue');
-        });
-
-        let newAction = select({
-            array: () => fooStore.array,
-        })(updateAction);
-
-        newAction();
-
-        expect(fooStore.array[0]).toBe('v');
-        expect(fooStore.array[1]).toBe('newValue');
-        expect(fooStore.array.length).toBe(2);
-    });
-
-    it('can update atoms', () => {
-        let fooStore = createStore('foo', {
-            array: ['v'],
-        })();
-
-        interface ActionState {
-            array: string[];
-        }
-
-        let updateAction = action('update')((state?: ActionState) => {
-            state.array[0] = 'newValue';
-        });
-
-        let newAction = select({
-            array: () => fooStore.array,
-        })(updateAction);
-
-        newAction();
-
-        expect(fooStore.array[0]).toBe('newValue');
-        expect(fooStore.array.length).toBe(1);
-    });
-
-    it('can update properties of an observable object', () => {
-        let fooStore = createStore('foo', {
-            obj: {
-                k: 'v',
-            },
-        })();
-
-        interface ActionState {
-            obj: { [key: string]: string };
-        }
-
-        let updateAction = action('update')((state?: ActionState) => {
-            /* tslint:disable:no-string-literal */
-            state.obj['k'] = 'newValue';
-            /* tslint:enable:no-string-literal */
-        });
-
-        let newAction = select({
-            obj: () => fooStore.obj,
-        })(updateAction);
-
-        newAction();
-
-        expect(fooStore.obj.k).toBe('newValue');
+        expect(readOnlyAction).toHaveBeenCalledTimes(1);
     });
 
     it('propagates action params to the selector function', () => {
         let fooStore: any = createStore('foo', {
             id0: 'value',
             array0: ['a', 'b', 'c'],
-        })();
```

Comment:
I think, with the changes @smikula made, you need the () on the end of all these createStore calls.